include/grpcpp/impl/codegen/client_interceptor.h

```diff
@@ -52,23 +53,43 @@ extern experimental::ClientInterceptorFactoryInterface*
 namespace experimental {
 class ClientRpcInfo {
  public:
-  ClientRpcInfo() {}
+  enum class Type {
+    UNARY,
+    CLIENT_STREAMING,
+    SERVER_STREAMING,
+    BIDI_STREAMING,
+    UNKNOWN
```

Comment:
Maybe `UNKNOWN` should come first, so it's the default value if things are memset to 0?

Reply:
I'm trying to avoid having to actually do any work in the static cast

Reply:
I should add that UNKNOWN only exists to support the default constructor in ClientRpcInfo, which is needed because ClientContext contains a ClientRpcInfo object. (see below for comparison to ServerContext)

Reply:
Okay.  How about adding a TODO about this?